### ISSUE TITLE
Add Gradient Clipping

### ISSUE THREAD

**Issue by A-Georgiou** [A-Georgiou/SmartDNN#30]
I did have gradient clipping but I removed it during some heavy debugging session, I do plan to add it back since it does really help stop exploding gradients during our backprop.

As we integrate with RNN's and deeper network architectures it will be required.
